File: src/main/java/at/hrechny/predictionsbot/controller/ServiceController.java
package at.hrechny.predictionsbot.controller;

import at.hrechny.predictionsbot.exception.interceptor.EnableErrorReport;
import at.hrechny.predictionsbot.model.PushUpdate;
import at.hrechny.predictionsbot.service.predictor.UserService;
import at.hrechny.predictionsbot.service.telegram.TelegramService;
import jakarta.validation.Valid;
import io.micronaut.http.MediaType;
import io.micronaut.http.HttpResponse;
import io.micronaut.http.annotation.Post;
import io.micronaut.http.annotation.Body;
import io.micronaut.http.annotation.Controller;

@Controller
@EnableErrorReport
public class ServiceController {

  private final UserService userService;
  private final TelegramService telegramService;

  public ServiceController(UserService userService, TelegramService telegramService) {
    this.userService = userService;
    this.telegramService = telegramService;
  }

  @Post(value = "/${secrets.adminKey}", consumes = MediaType.APPLICATION_JSON)
  public HttpResponse<Void> pushUpdate(@Valid @Body PushUpdate pushUpdate) {
    userService.getUsers().forEach(user -> telegramService.pushUpdate(user.getId(), pushUpdate.getMessage(), pushUpdate.isUpdateCompetitionList()));
    return HttpResponse.ok();
  }

}

File: src/main/kotlin/at/hrechny/predictionsbot/controller/ServiceController.kt
package at.hrechny.predictionsbot.controller

import at.hrechny.predictionsbot.controller.model.service.PushUpdateRequestDto
import at.hrechny.predictionsbot.exception.interceptor.EnableErrorReport
import at.hrechny.predictionsbot.service.predictor.UserService
import at.hrechny.predictionsbot.service.telegram.TelegramService
import io.micronaut.http.HttpResponse
import io.micronaut.http.MediaType
import io.micronaut.http.annotation.Body
import io.micronaut.http.annotation.Controller
import io.micronaut.http.annotation.Post
import jakarta.validation.Valid

@Controller
@EnableErrorReport
open class ServiceController(
    private val userService: UserService,
    private val telegramService: TelegramService,
) {
    @Post(value = "/\${secrets.adminKey:}", consumes = [MediaType.APPLICATION_JSON])
    open fun pushUpdate(@Valid @Body pushUpdate: PushUpdateRequestDto): HttpResponse<Void> {
        userService.getUsers().forEach { user ->
            telegramService.pushUpdate(user.id, pushUpdate.message, pushUpdate.isUpdateCompetitionList())
        }
        return HttpResponse.ok()
    }
}
